### ISSUE TITLE
Compilation fails without the --mingw64 switch.

### ISSUE THREAD

**Issue by golyalpha** [Nuitka/Nuitka#396]
Nuitka installed from pip through Poetry, in a Poetry-managed virtual environment:
```
>py -mnuitka --version
0.6.3.1
Python: 3.7.0 (v3.7.0:1bf9cc5093, Jun 27 2018, 04:59:51) [MSC v.1914 64 bit (AMD64)]
Executable: C:\Users\golya\AppData\Local\pypoetry\Cache\virtualenvs\[project_name]-py3.7\Scripts\python.exe
OS: Windows
Arch: x86_64
```
I created the test file as provided in the user manual, and tried to build it with the following command:
```
>py -mnuitka --standalone --show-progress --show-scons test.py
```
Here's the scons output: 
[scons_output.txt](https://github.com/Nuitka/Nuitka/files/3202290/scons_output.txt)

I have both VS2017 and MingW64 installed (scons was crashing without VS2017 installed).
Adding the option `--mingw64` compiles a working binary, but I feel like that shouldn't be a requirement.

**Comment by kayhayen** [comment]
I would have been interested in the crash output for no MSVC. I think develop or factory branch, one of which now autodetects `--mingw` from the command line, you are supposed for stable right now, to specify it, otherwise you CC is assumed to be a `cl.exe`.

That is going to improve the out of the box experience, until then I believe setting `PATH` appropiately may work more automatically, not sure right now though.

Yours,
Kay

**Comment by kayhayen** [comment]
Btw, what is Poetry? :)

**Comment by kayhayen** [comment]
So this is in the last pre-release, setting a CC will now infer --mingw64 mode automatically from the name it has.

**Comment by golyalpha** [comment]
[Poetry](https://github.com/sdispater/poetry/), it's similar Pipenv, but some people like it more, for various reasons. I use it for pretty much every project that doesn't deploy on places that don't support it.  

Do I understand correctly, that you'd like to see the crash as it was without VS2017 installed?

**Comment by kayhayen** [comment]
@golyalpha yes indeed, crash is something that shouldn't happen, but it's near impossible for me to remove it from my machines, so I am dependent on reports there.

Also looked at Poetry. Interesting. I am still a `pipenv` newbie though, sadly I cannot use it for Nuitka, e.g. because parallel maintenance of the many Python versions supported is not foreseen. What I found strange is that they do not recommend `pip install poetry` for installing it.

**Comment by golyalpha** [comment]
Yeah. They have their own install script for Poetry. From what I've tried, it installs fine through pip. I personally just install it through pip when running on pipelines, etc. Installing from pip breaks the self update functionality though, and I think the author mentions something about package conflicts...?

**Comment by kayhayen** [comment]
Released as 0.6.4

**Comment by golyalpha** [comment]
@kayhayen Sorry I didn't send the log yet, been in the process of switching computers. Were you able to reproduce the issue with the VS2017 not installed, or should I open a new issue for that?
